### DISCUSSION TITLE
Memory leak and/or slow tests

### DISCUSSION THREAD

**Discussion by Talinon** [livewire/livewire#7488] [7. Bugs (archived)]
### Livewire version

3.2.6

### Which PHP version are you using?

PHP 8.2

### Steps To Reproduce

Run a large test suite, or create a test that loops many times and dump out the memory usage

example:
```
public function test_memory_usage() {

    for($i = 1; $i <= 1000; ++$i) {

        $this->createApplication()->flush();
        
        dump('Using ' . ((int) (memory_get_usage(true) / (1024 * 1024))) . 'MB.  Iteration: ' . $i);
    }
}
```

With Livewire bootstrapped: Using 497MB.  Iteration: 1000
Without: Using 112MB.  Iteration: 1000

### Additional information

I went on a wild goose chase after installing the new ``laravel/pulse`` first party package. Immediately after installing, my test suite took over 3 times as long to run. Approximately 2000 tests went from taking 3 minutes to over 10 minutes to complete.

I first suspected it was something in the pulse beta version, but after a process of elimination, I eventually discovered it was when the Livewire package is installed (dependency of Pulse)

I understand that as PHPUnit runs sequential tests, it doesn't clean up after itself and certain things are not cleared from memory, causing usage to grow. However, when Livewire is installed, the memory being consumed is exponentially more.

I traced to the ``bootMechanisms()`` method on the LivewireServiceProvider at this point:
https://github.com/livewire/livewire/blob/f861e8af7d08b388ea77553d7f4e382830001277/src/LivewireServiceProvider.php#L77

Commenting out that line and re-running the test suite results in the tests completing more than three times faster.

My workaround for solving this was to:

1. Disable Livewire auto-discovery in composer.json:
```
    "extra": {
        "laravel": {
            "dont-discover": [
                "livewire/livewire"
            ]
        }
    },
```

2. Conditionally load LivewireServiceProvider depending upon the application environment within AppServiceProvider:
```
   if ($this->app->environment() != 'testing') {
      $this->app->register(\Livewire\LivewireServiceProvider::class);
   }

```

I'm not sure if there is anything practical that can be done about this, but just bringing it to your attention. I'm glad I was able to come up with a work around, because I wasn't willing to have my test suite take 3 times longer. 

Keep up the good work with Livewire!

Thanks

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by joshhanley** [comment]
@Talinon please see PR #7181 for more details about this issue
